compiler/backend/src/org/jetbrains/kotlin/codegen/JvmCodegenUtil.java

```diff
@@ -199,15 +199,27 @@ public static boolean couldUseDirectAccessToProperty(
         if (KotlinTypeMapper.isAccessor(property)) return false;
 
         CodegenContext context = contextBeforeInline.getFirstCrossInlineOrNonInlineContext();
-        // Inline functions can't use direct access because a field may not be visible at the call site
-        if (context.isInlineMethodContext()) {
+        // Inline functions or inline classes can't use direct access because a field may not be visible at the call site
+        if (context.isInlineMethodContext() || (context.getEnclosingClass() != null && context.getEnclosingClass().isInline())) {
```

Comment:
It is mandatory otherwise test `testReferToPropertyInCompanionObjectOfInlineClass` will failed
because when `couldUseDirectAccessToProperty` is used all conditions are satisfied but later the code accessing the private property into this test is moved into a class suffixed by `$Erased` and it is not longer legal to use a direct access if the access is move into another context.